src/service/mobile_config.proto

```diff
@@ -65,36 +65,34 @@ message gateway_info_stream_res_v1 {
   bytes signature = 4;
 }
 
-message router_get_req_v1 {
-  // Pubkey binary of the requested router
-  bytes pubkey = 1;
-  // pubkey binary of the signing keypair
-  bytes signer = 2;
-  // Signature over the request by the requesting oracle
-  bytes signature = 3;
+enum admin_key_role {
+  // administrative operator key
+  administrator = 0;
+  // routing infrastructure key for routing streams
+  router = 1;
+  // keys for verifying requests from other oracles
+  oracle = 2;
+  // carrier authorizing keys for signing mobile subscriber activity
+  carrier = 3;
 }
 
-message router_get_res_v1 {
-  // Pubkey binary of the requested router, confirming registration
-  bytes pubkey = 1;
-  // unix epoch timestamp in seconds
-  uint64 timestamp = 2;
-  // pubkey binary of the signing keypair
-  bytes signer = 3;
-  // Signature over the response by the config service
-  bytes signature = 4;
+enum network_key_role {
+  carrier = 0;
+  router = 1;
 }
 
-message router_list_req_v1 {
+message authorization_list_req_v1 {
+  // role of the keys being requested
+  network_key_role role = 1;
   // pubkey binary of the signing keypair
-  bytes signer = 1;
+  bytes signer = 2;
   // Signature over the request by the requesting oracle
-  bytes signature = 2;
+  bytes signature = 3;
 }
 
-message router_list_res_v1 {
-  // List of public key binaries of all registered routers
-  repeated bytes routers = 1;
+message authorization_list_res_v1 {
+  // List of public key binaries of all registered entities by requested role
+  repeated bytes entity = 1;
```

Comment:
why use entity here when elsewhere pubkey is in use, ie when a key is added its called a pubkey but when listed its an entity

Reply:
fixed [c71ed9c](https://github.com/helium/proto/pull/340/commits/c71ed9c16fc1efdff1d2aed1294f1f47dd93c8bc)